geodata/models/raster/reader.py

```diff
@@ -0,0 +1,128 @@
+"""Helper methods for creating a ``GDALRaster`` entry from a raster file."""
+from celery.utils.log import get_task_logger
+from django.contrib.gis.gdal import SpatialReference
+from django.contrib.gis.geos import Polygon
+import rasterio
+
+# from django.core.files.base import ContentFile
+# import io
+# from PIL import Image
+
+from .base import BandMetaEntry, ConvertedRasterFile, RasterEntry, RasterFile
+from ..common import _ReaderRoutine
+
+logger = get_task_logger(__name__)
+
+
+class RasterEntryReader(_ReaderRoutine):
+    """Raster injestion routine.
+
+    This helper class will open a raster file and create the ``GDALRaster``
+    for the ``RasterEntry``'s raster property.
+
+    """
+
+    # @staticmethod
+    # def create_thumbnail(src):
+    #     oview = int(max(src.height, src.width) * 0.1) or 1
+    #     thumbnail = src.read(1, out_shape=(1, int(src.height // oview), int(src.width // oview)))
+    #
+    #     buf = io.BytesIO()
+    #     thumbnail.save(buf, format='JPEG')
+    #     byte_im = buf.getvalue()
+    #     return ContentFile(byte_im)
+
+    def _read_files(self):
+        """Load raster data files and create a ``RasterEntry``.
+
+        This function will need to handle a lot of scenarios for how
+        the raster file is passed and where it is stored.
+
+        """
+        # Fetch the raster file this Layer corresponds to
+        self.rfe = RasterFile.objects.get(id=self.model_id)
+        file_path = self.rfe.raster_file.name
```

Comment:
Because we are using S3/minio, we can't just do this.  Rather, we should use the `_field_file_to_local_path` context function (and that function should get moved to some common place to avoid duplication in core).  If that function is available, then this would be
```
with _field_file_to_local_path(self.rfe.raster_file) as file_path:
   <all following code>
```

Reply:
I pushed a commit that does this.